src/main/java/com/music/wynk/services/SongServiceImpl.java

```diff
@@ -2,48 +2,63 @@
 
 import com.music.wynk.models.Song;
 import com.music.wynk.repositories.SongRepository;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @Service
 public class SongServiceImpl implements SongService {
 
+    @Autowired
     private SongRepository songRepository;
 
-    public SongServiceImpl(SongRepository songRepository) {
-        this.songRepository = songRepository;
-    }
-
     @Override
-    public List<Song> getSongs() {
-        return songRepository.findAll();
+    public List<Song> getSongs(boolean flag) {
+        List<Song> songs = new ArrayList<>();
+        songRepository.findAll().forEach(songs::add);
+        if (flag) {
+            songs.sort((Song s1, Song s2) -> s2.getSongPopularity()-s1.getSongPopularity());
```

Comment:
Repository is not giving me back a list. It is just giving a iterable so that's why it was needed to cast it to a list to perform sort operation and return a list to controller